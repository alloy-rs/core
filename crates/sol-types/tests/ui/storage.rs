use alloy_sol_types::sol;

sol! {
    struct BadMemoryStruct {
        bool memory a;
    }
}

sol! {
    struct BadStorageStruct {
        bool storage a;
    }
}

sol! {
    struct BadCalldataStruct {
        bool calldata a;
    }
}

sol! {
    function badMemoryFunction(
        bool memory a,
    );
}

sol! {
    function badStorageFunction(
        bool storage a,
    );
}

sol! {
    function badCalldataFunction(
        bool calldata a,
    );
}

sol! {
    struct Custom {
        bool a;
    }

    function validStorage(
        bytes memory a,
        bytes storage b,
        bytes calldata c,

        string memory d,
        string storage e,
        string calldata f,

        bool[] memory g,
        bool[] storage h,
        bool[] calldata i,

        Custom memory j,
        Custom storage k,
        Custom calldata l,
    ) external;
}

fn main() {}
