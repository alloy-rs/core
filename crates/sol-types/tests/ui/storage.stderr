error: data locations are not allowed in struct definitions
 --> tests/ui/storage.rs:5:14
  |
5 |         bool memory a;
  |              ^^^^^^

error: data locations are not allowed in struct definitions
  --> tests/ui/storage.rs:11:14
   |
11 |         bool storage a;
   |              ^^^^^^^

error: data locations are not allowed in struct definitions
  --> tests/ui/storage.rs:17:14
   |
17 |         bool calldata a;
   |              ^^^^^^^^

error: data location can only be specified for array, struct or mapping types
  --> tests/ui/storage.rs:23:14
   |
23 |         bool memory a,
   |              ^^^^^^

error: data location can only be specified for array, struct or mapping types
  --> tests/ui/storage.rs:29:14
   |
29 |         bool storage a,
   |              ^^^^^^^

error: data location can only be specified for array, struct or mapping types
  --> tests/ui/storage.rs:35:14
   |
35 |         bool calldata a,
   |              ^^^^^^^^
